main/src/ca/uwaterloo/flix/language/phase/inference/RestrictableChooseInference.scala

```diff
@@ -216,12 +216,13 @@ object RestrictableChooseInference {
           enumSym,
           loc
         )
-        _ <- unifySubset(indexOutVar, set, enumSym, loc)
+//        _ <- unifySubset(indexOutVar, set, enumSym, loc)
+        _ <- unifySubset(set, indexOutVar, enumSym, loc)
 
         resultTconstrs = constrs ::: constrss.flatten
 
         // τ_out
-        resultTpe <- unifyTypeM(tpe0 :: tpes, loc)
+        resultTpe <- unifyTypeM(enumTypeOut, tpe0, loc)
```

Comment:
And this too I guess?

Reply:
No this was just a bug bug that made the unification and reconstruction inconsistent